docs/android/testing/integration_testing.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "Android integration testing"
+sidebar_label: "Integration testing"
+---
+
+🚧🚧🚧 Under Construction 🚧🚧🚧
+
+## Why perform integration testing?
+
+[Unit tests](unit_testing) are great and should be your primary choice when writing tests. However, integration testing ensures that the behavior of the application in a real Android environment is validated. Integration tests run on a real Android OS through an emulator, using the same JVM that will be used by end users.
+
+### Testing on a real JVM
+
+Some libraries we use have specific requirements, such as a minimum Java version above 8. Since our application targets Android API 21, which uses Java 8, these libraries cannot always be used. 
+
+During development, you might only test on the latest Android OS version, which uses a Java version above 8. If no one catches this during the review process, users on Android API 21 may encounter crashes.
+
+A concrete example is the [Jackson library](https://github.com/FasterXML/jackson). Starting from version 2.14, it requires a minimum Android version of 26. Unfortunately, such errors only arise at runtime, so the only way to catch them is through instrumentation tests. You can see the concrete example on this [PR](https://github.com/home-assistant/android/pull/5108).
+
+### UI or no UI?
+
+Integration tests do not always involve displaying a UI. They are also used to test [foreground services](https://developer.android.com/develop/background-work/services/fgs), where no UI is displayed. In these cases, we only verify the data and the interaction with the system.
+
+### Testing with the corresponding Home Assistant Core version
+
+We aim to run integration tests against the corresponding version of Home Assistant Core. This ensures that the current code works seamlessly with the core version in a straightforward manner.
```

Comment:
What is the corresponding version? The app is built to be independent of the core version because we cannot control updates (availability of the app, the user updating app/core, aligning version numbers). If a core version requirement exists, the app should contain a check for it.

Reply:
This part is something that I want to work on pretty soon. My plan is to actually run the test against the stable version of HA to make sure nothing is broken. It's the end-to-end tests.

Reply:
OK, but you still cannot easily point to a "corresponding" version. Suggestion: against the latest version of Home Assistant Core.

Reply:
I will use your suggestion, but I think we need to be smarter when writing the test while selecting the version. Let's talk about it in another place.